### PR TITLE
Added the ability to pass a config_command to config_mode when callin…

### PR DESCRIPTION
**Pull request by autonomouse**
…g send_config_set

### CODE REVIEW

**Comment by autonomouse** [comment]
Should have left a comment, sorry. So we're basically tying to use "configure private" for juniper, but I've noticed that the `output = self.config_mode()` in `send_config_set` will overwrite any external use of `netmiko_instance.config_mode(config_command='configure private')`.

File: netmiko/base_connection.py [hunk 1]
```diff
@@ -949,7 +949,8 @@ def send_config_from_file(self, config_file=None, **kwargs):
             return self.send_config_set(cfg_file, **kwargs)
 
     def send_config_set(self, config_commands=None, exit_config_mode=True, delay_factor=1,
-                        max_loops=150, strip_prompt=False, strip_command=False):
+                        max_loops=150, strip_prompt=False, strip_command=False,
+                        config_mode_cmd=None):
```
Comment (markgoddard) [inline netmiko/base_connection.py]: Perhaps use config_command or config_mode_command for consistency?

File: netmiko/base_connection.py [hunk 2]
```diff
@@ -968,7 +969,7 @@ def send_config_set(self, config_commands=None, exit_config_mode=True, delay_fac
             raise ValueError("Invalid argument passed into send_config_set")
 
         # Send config commands
-        output = self.config_mode()
+        output = self.config_mode(config_mode_cmd) if config_mode_cmd else self.config_mode()
```
Comment (markgoddard) [inline netmiko/base_connection.py]: Might be better to do this with a kwargs pattern:
```
kwargs = {}
if config_mode_cmd:
    kwargs['config_command'] = config_mode_cmd
output = self.config_mode(**kwargs)
```
Something similar could be achieved with positional args.

**Comment by ktbyers** [comment]
I will modify this slightly after merged.

**Pull request merged**